### DISCUSSION TITLE
SFCC TO SFMC Migration (SFMC integration in Pipeline based Project)

### DISCUSSION THREAD

**Discussion by ganeshbadiga** [SalesforceCommerceCloud/sfcc-ci#549] [Team Posts]
Hi Team,
 How to connect Sales force commerce cloud to Sales force Marketing cloud. Currently we are working on pipeline based project. Could you please suggest how i can implement. where can i get pipeline based Salesforce Marketing Cloud cartridge or documents.

Thanks
B Ganesh

**Comment by philipstyron-zz** [comment]
I've integrated with MC for an OMS implementation recently.

For that, I wrote a Digital job that would run as frequently as you need.
Each time, it would do these things:

- Select any new objects for which you want to send an email
- Get the detail information for that object and use it to build the MC
request JSON
- Post the JSON request to MC Send API
<https://developer.salesforce.com/docs/atlas.en-us.noversion.mc-apis.meta/mc-apis/messageDefinitionSends.htm>
.

NOTE: The MC message had to be pre-configured and set up within MC, and
custom JSON parameters defined

Hope this helps. I can send you the code for one of my requests if that
would be helpful.

Thank you,

Philip Styron
Technical Architect Director | Salesforce
Mobile: 512-629-6186
<http://smart.salesforce.com/sig/pstyron//us_mb_kb/default/link.html>

**Comment by ganeshbadiga** [comment]
Hi Philip Styron,

Thanks for your help. I got some idea. actually I need to implement Email Marketing Functionality (signup, 
 sendtofriend, abandoned  cart, order confirmation) Through Marketing Cloud. Can i use below cartridge https://github.com/SalesforceCommerceCloud/marketing-cloud-connector (or) any other document/cartridge. suppose if i have to use above cartridge means i need to convert it to pipelines. is it support for pipeline version.

Could you please assist me.

Thanks 
B Ganesh

**Comment by zlatinz** [comment]
Hi,
You can mix controllers and pipelines in your solution.
There are situations where you cannot use controllers directly, for example if you want to use a jump node from the pipeline that is to call a controller. This is an example of a thing you cannot do. 
In cases like this, you need to create additional wrapper code as pipeline and scripts that use the existing functionality of the controller version.

I do not know if any customer has already done this, but this should not be complicated for an experienced developer or a technical architect.

Hope this helps,
Zlatin

**Comment by pravngaur** [comment]
Hello,

This will not work for pipelines based implementations as the connector
cartridge uses the handler framework which has dependency on controller(s).
So, ootb this will not work with pipeline.

But even your storefront code is in pipeline, the connector code may still
br in controlers. Do you see an issue there?

**Comment by bahenry** [comment]
You can also look into the deprecated int_exacttarget integration for historical view of a pipeline implementation and sending payloads to custom triggers. 

Please be aware you are emailing everyone added to the SFCC team

Bryte

**Comment by ganeshbadiga** [comment]
Hi Bryte,

Thank you for your information. Going forward I will send queries either private or in slack. Thank you team.

Thanks
B Ganesh
